src/azure-cli/azure/cli/command_modules/vm/custom.py

```diff
@@ -2789,13 +2801,26 @@ def create_image_version(cmd, resource_group_name, gallery_name, gallery_image_n
     if cmd.supported_api_version(min_api='2019-07-01', operation_group='gallery_image_versions'):
         GalleryImageVersionStorageProfile = cmd.get_models('GalleryImageVersionStorageProfile')
         GalleryArtifactVersionSource = cmd.get_models('GalleryArtifactVersionSource')
-        # GalleryOSDiskImage = cmd.get_models('GalleryOSDiskImage')
-        # GalleryDataDiskImage = cmd.get_models('GalleryDataDiskImage')
-        source = GalleryArtifactVersionSource(id=managed_image)
-        storage_profile = GalleryImageVersionStorageProfile(source=source)
+        GalleryOSDiskImage = cmd.get_models('GalleryOSDiskImage')
+        GalleryDataDiskImage = cmd.get_models('GalleryDataDiskImage')
+        source = os_disk_image = data_disk_images = None
+        if managed_image is not None:
+            source = GalleryArtifactVersionSource(id=managed_image)
+        if os_snapshot is not None:
+            os_disk_image = GalleryOSDiskImage(source=GalleryArtifactVersionSource(id=os_snapshot))
+        if data_snapshots != []:
+            data_disk_images = []
+            for i, s in enumerate(data_snapshots):
+                data_disk_images.append(GalleryDataDiskImage(source=GalleryArtifactVersionSource(id=s), lun=i))
+        if managed_image is None and os_snapshot is None:
+            raise CLIError('usage error: Please provide --managed-image or --os-snapshot')
+        storage_profile = GalleryImageVersionStorageProfile(source=source, os_disk_image=os_disk_image,
+                                                            data_disk_images=data_disk_images)
         image_version = ImageVersion(publishing_profile=profile, location=location, tags=(tags or {}),
                                      storage_profile=storage_profile)
     else:
+        if managed_image is None:
```

Comment:
I cannot find the usage of `managed_image`. Why do you need to check it?

Reply:
> 
> 
> I cannot find the usage of `managed_image`. Why do you need to check it?

It's used in profile.